### PR TITLE
contact the POCs from a listing

### PR DESCRIPTION
**Pull request by flesher1**
#866 

requires https://github.com/aml-development/ozp-backend/pull/358

Add ability to notify from Listing menu
 - Adds button to the details tab on listing page
 - Adds (review like) text box to fill in notificaiton text
 - Sends notification using API ListingOwnerNotification
 - Requires send_notificaiton_to_owner branch

closes ticket #866 in GIT

### CODE REVIEW

File: app/js/components/quickview/DetailsTab.jsx [hunk 1]
```diff
@@ -21,7 +22,19 @@ var DetailsTab = React.createClass({
...
           refresh: false
         };
-      },
+    },
+
+    getInitialState: function () {
+        return this.getState();
+    },
+
+    getState: function () {
```
Comment (mleeBoeing) [inline app/js/components/quickview/DetailsTab.jsx]: You can just move this function into getInitialState

File: app/js/components/quickview/DetailsTab.jsx [hunk 2]
```diff
@@ -21,7 +22,19 @@ var DetailsTab = React.createClass({
...
+        var currentUser = SelfStore.getDefaultData().currentUser;
```
Comment (mleeBoeing) [inline app/js/components/quickview/DetailsTab.jsx]: Should already have currentUser available in props

File: app/js/components/quickview/NotifyOwners.jsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+
+    propTypes: {
+        listing: React.PropTypes.object.isRequired,
+        text: React.PropTypes.string.isRequired,
+    },
+
+    getDefaultProps: function() {
+        return {
+            text: ''
```
Comment (mleeBoeing) [inline app/js/components/quickview/NotifyOwners.jsx]: Can remove

File: app/js/components/quickview/NotifyOwners.jsx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        }
+    },
+
+    getInitialState: function () {
+        return {
+            uuid: uuid(),
+            date: null,
```
Comment (mleeBoeing) [inline app/js/components/quickview/NotifyOwners.jsx]: Can remove

File: app/js/components/quickview/NotifyOwners.jsx [hunk 5]
```diff
@@ -0,0 +1,101 @@
+'use strict';
+
+var React = require('react');
+var Reflux = require('reflux');
+var _ = require('../../utils/_');
+var uuid = require('../../utils/uuid.js');
+var NotificationActions = require('../../actions/NotificationActions.js');
+
+var SubmitResponse = React.createClass({
```
Comment (mleeBoeing) [inline app/js/components/quickview/NotifyOwners.jsx]: Class and filename does not match

File: app/js/components/quickview/NotifyOwners.jsx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+        }
+    },
+
+    render: function() {
+        var { text } = this.state;
+        var submitButton = text.length > 0 ? <button className="btn btn-success btn-small" onClick={ this.onSave } >Submit</button> : <button className="btn btn-success btn-small" onClick={ this.onSave } disabled>Submit</button>;
+
+        return (
+            <div>
+                <h6>Write a text to the above owners</h6>
```
Comment (blynch11p) [inline app/js/components/quickview/NotifyOwners.jsx]: Needs better verbage.  
"Send a message|note|notice to the above owners"
"Write text to the above owners"
"Contact above owners"

**Comment by blynch11p** [comment]
I'm getting a console error in firefox after clicking Submit: `ReferenceError: event is not defined`

The notification sends, but the input box doesn't go away, somewhat implying to the user that the notification did not send.

**Comment by mleeBoeing** [comment]
Adds button to the details tab on listing page :+1:
Adds (review like) text box to fill in notificaiton text :+1:
Sends notification using API ListingOwnerNotification :+1:
Tested on Chrome

Found a bug. Steps to reproduce:
1. Login as bigbrother
2.. Open Bass Fishing
3. Send notification to 2 owners, should be Betta Fish and Jones
4. Log in with bettafish and jones to verify the notifications
5. If you also login with wsmith, you'll see the same notification even though wsmith is not listed under the owners.

This is most likely a backend issue though.

**Comment by flesher1** [comment]
I should have included more information than I did on what this notification thing does. 

To test, you can follow the example above.

Login as bigbrother
 Open Bass Fishing
Send notification to 2 owners, should be Betta Fish and Jones
Log in with bettafish and jones to verify the notifications
Login with wsmith, wsmith is a steward for minitruth and is sent a copy of the notification.

Would this be something we want to notify the user of? I was going to silently include the Steward on the message instead of telling the user.We can include the info in the sentence above the text box.

**Comment by blynch11p** [comment]
The ticket still specifies that this Notification be sent from the Launch Notice (https://github.com/aml-development/ozp-center/issues/866), so, while, so far in my testing the Notification being sent from the Details tab works, the PR doesn't meet the ticket's purpose.

**Comment by blynch11p** [comment]
A possible alternative to my comment above would be to add a note in the Launch Notice saying something along the lines of:
'If you have any issues launching this application, view the <link to Details tab> or <link to Resources tab> for information on how to contact Application POCs'

**Comment by flesher1** [comment]
I added a button on the Leaving OZP popup. This will take the user to the notification submission area in the listing.

**Comment by blynch11p** [comment]
This is pretty good.  I might recommend a change to the layout of the popup a little.
layout_change

And make the button look 'active'.  I think `class="confirm"` is what's used for the popup blocked button.

**Comment by lsemesky** [comment]
@flesher1 this is ready to review?

**Comment by blynch11p** [comment]
Owners of Applications can be contacted from the Details tab
A way to contact owners, by sending the user to the Details tab, is available in the Listing Notice dialog.

👍

**Comment by rkenyon1969** [comment]
1. pulled branch(es)
2. reviewed code
3. tested

:+1:

**Pull request merged**